src/viewport/background_tasks.py

```
@@ -388,6 +435,534 @@ def create_thumbnails_batch_task(self, photos: list[ThumbnailTaskPayload]) -> di
 
+# --- Video processing ---------------------------------------------------------
+
+MAX_VIDEO_DURATION_SECONDS = 1800
+
+
+def _ffprobe_streams(filepath: str) -> dict[str, Any]:
+    """Run ffprobe and return parsed JSON stream metadata.
+
+    Raises subprocess.CalledProcessError on ffprobe failure.
+    """
+    result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)